cfme/tests/containers/test_cockpit.py

```diff
@@ -11,25 +12,35 @@
 
 
 @pytest.mark.polarion('CMP-10255')
-def test_cockpit_button_access(provider, appliance, soft_assert):
+# Cockpit Feature is only available in 5.9 and greater
+@pytest.mark.uncollectif(lambda appliance, provider: appliance.version <= "5.9")
+@pytest.mark.parametrize('cockpit', [False, True], ids=['disabled', 'enabled'])
+def test_cockpit_button_access(appliance, provider, cockpit):
     """ The test verifies the existence of cockpit "Web Console"
-        button on master node, then presses on the button and
-        opens up the cockpit main page in a new window. Then
-        we verify the title of the main cockpit page. The test
-        will not work until the single sign-on bug is fixed.
+        button on master node, click the button if enabled, verify no errors are displayed.
     """
 
+    if cockpit:
+        appliance.server.settings.enable_server_roles('cockpit_ws')
+        wait_for(lambda: appliance.server_roles['cockpit_ws'] is True, delay=10, timeout=300)
+    elif not cockpit:
+        appliance.server.settings.disable_server_roles('cockpit_ws')
+        wait_for(lambda: appliance.server_roles['cockpit_ws'] is False, delay=10, timeout=300)
+    else:
+        pytest.skip("Cockpit should be either enabled or disabled.")
+
     collection = appliance.collections.container_nodes
     nodes = collection.all()
     node = [node for node in nodes if 'master' in node.name][0]
     view = navigate_to(node, 'Details')
 
-    assert view.toolbar.web_console.is_displayed, 'No "Web Console" button found'
-    assert view.toolbar.web_console.active
-    view.toolbar.web_console.click()
-
-    port_num = ':9090/system'
-    url_cockpit = provider.hostname + port_num
-    view.browser.url = url_cockpit
-    title_pg = view.browser.selenium.title()
-    assert title_pg == 'Cockpit', 'Cockpit main page failed to open'
+    if cockpit:
+        appliance.server.browser.refresh()
+        assert not view.toolbar.web_console.disabled
+        view.toolbar.web_console.click()
```

Comment:
I don't like completely losing the verification that at least the new console tab was opened, and I think this creates some bad assumptions for validation - if the `click()` did open a new tab, the flash widget won't be displayed to call `assert_no_error` against it.

I'm thinking a flow like the following pseudo-code would be a more deterministic verification:
```
view.toolbar.web_console.click()
if provider.version < 3.6:
    assert view.is_displayed  # shouldn't have changed
    view.flash.assert_message('Feature not available/supported')
else:
    assert not view.is_displayed  # browser should be on a different tab now
    assert ':9090/system' in view.browser.url
    assert provider.hostname in view.browser.url  # may not work if the IP is used
    # close the console tab
    assert view.is_displayed
    view.flash.assert_no_error()
```